freshmaker/lightblue.py

```diff
@@ -1576,8 +1576,12 @@ def _get_images_to_rebuild(image):
         to_rebuild = self._deduplicate_images_to_rebuild(to_rebuild)
         # Get all the directly affected images so that any parents that are not marked as
         # directly affected can be set in _images_to_rebuild_to_batches
+        # And in addition mark all latest EUS images as directly affected so
+        # botas could correctly file an advisory
+        repos = self.find_all_container_repositories(published, release_categories)
```

Comment:
`published` would need to be `False` to find the EUS image repositories. That being said, this is essentially duplicating the query in `find_images_with_packages_from_content_set` and it is quite expensive.

The `image` dictionary will have the `repositories` key, so you can search for the auto-rebuild tags of those repositories and see if the EUS image currently has that tag. I would also convert `is_latest_eus_image` to a method on this class and then have it cache the autorebuilds tags for the EUS repositories on the object.

Reply:
So if I got you right, you suggest to make a request to LB for every repo in `repositories' of every image and then get `auto_rebuild_tags` from the response. And check if any of this tags corresponds to the tags that are in image's `repositories' and if so mark this image as **directly_affected**?

Reply:
Yes, but when you say "every image", it would be every EUS image.